### PR TITLE
feat(transformer/nullish-coalescing-operator): add comments in top of file

### PR DESCRIPTION
**Pull request by Dunqing**
part of #4881

### CODE REVIEW

**Comment by Dunqing** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4972?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4973** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4973?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4972** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4972?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4965** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4965?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by overlookmotel — APPROVED** [review]
I've pushed 4 commits with suggested changes. I've only done it as commits as it's hard to do this as review comments. Please take them as suggestions, not demands!

These changes are all nitty and not very important, but as we're trying to set a standard, and this is the first PR, I thought it worthwhile to bikeshed even the small stuff.

Again, these suggested changes are just my opinion of what makes for maximum clarity for a reader. But in my view it's less important what our "style" is, than that we're all happy with it, and that we use it uniformly. So if you think it's too wordy, or is unclear, please do feel free to revert my changes, or amend further.

**Comment by Dunqing** [comment]
@overlookmotel Thanks! Looks so clear!

**Review by DonIsaac — APPROVED** [review]
